tests/bsim/bluetooth/host/central/src/main.c

```diff
@@ -82,25 +101,118 @@ static void test_central_connect_timeout(void)
+	TEST_ASSERT(err == -EALREADY, "Expected to fail to create connection (err %d)", err);
+
+	/* Expect the number of refs to be unchanged. */
+	TEST_ASSERT(atomic_get(&conn->ref) == initial_refs,
+		    "Expect number of references to be unchanged");
+
+	err = k_sem_take(&sem_connected, K_FOREVER);
+	TEST_ASSERT(err == 0, "Failed getting connected timeout", err);
+
+	TEST_ASSERT(atomic_get(&conn->ref) == 0, "Expect no more references");
+
+	TEST_PASS("Passed");
+}
+
+static void test_central_connect_to_existing(void)
+{
+	int err;
+
+	bt_conn_cb_register(&conn_cb);
+
+	/* Initialize Bluetooth */
+	err = bt_enable(NULL);
+	TEST_ASSERT(err == 0, "Can't enable Bluetooth (err %d)", err);
+
+	struct bt_conn *conn;
+
+	bt_addr_le_t peer = {.type = BT_ADDR_LE_RANDOM,
+			     .a.val = {0xc0, 0xc0, 0xc0, 0xc0, 0xc0, 0xc0}};
+
+	const struct bt_conn_le_create_param create_param = {
+		.options = BT_CONN_LE_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	k_sem_reset(&sem_connected);
+
+	err = bt_conn_le_create(&peer, &create_param, BT_LE_CONN_PARAM_DEFAULT, &conn);
+	TEST_ASSERT(err == 0, "Failed starting initiator (err %d)", err);
+
+	err = k_sem_take(&sem_connected, K_FOREVER);
+	TEST_ASSERT(err == 0, "Failed establishing connection", err);
+
+	/* Now we have a valid connection reference */
+	atomic_val_t initial_refs = atomic_get(&conn->ref);
+
+	TEST_ASSERT(initial_refs >= 1, "Expect to have at least once reference");
+
+	err = bt_conn_le_create(&peer, &create_param, BT_LE_CONN_PARAM_DEFAULT, &conn);
+	TEST_ASSERT(err == -EINVAL, "Expected to fail to create a connection (err %d)", err);
+
+	/* Expect the number of refs to be unchanged. */
+	TEST_ASSERT(atomic_get(&conn->ref) == initial_refs,
+		    "Expect number of references to be unchanged");
+
+	TEST_PASS("Passed");
+}
+
 static const struct bst_test_instance test_def[] = {
-	{
-		.test_id = "central_connect_timeout",
-		.test_descr = "Verifies that the default connection timeout is used correctly",
-		.test_tick_f = bst_tick,
-		.test_main_f = test_central_connect_timeout
-	},
-	BSTEST_END_MARKER
-};
+	{.test_id = "central_connect_timeout",
+	 .test_descr = "Verifies that the default connection timeout is used correctly",
+	 .test_tick_f = bst_tick,
+	 .test_main_f = test_central_connect_timeout},
+	{.test_id = "central_connect_when_connecting",
+	 .test_descr = "Verifies that the stack returns an error code when trying to connect"
+		       " while already connecting",
+	 .test_tick_f = bst_tick,
+	 .test_main_f = test_central_connect_when_connecting},
+	{.test_id = "central_connect_to_existing",
+	 .test_descr = "Verifies that the stack returns an error code when trying to connect"
+		       " to an existing device and does not unref the existing connection object.",
+	 .test_tick_f = bst_tick,
+	 .test_main_f = test_central_connect_to_existing},
+	BSTEST_END_MARKER};
```

Comment:
Ditto with the trailing commas (not a needed change as our coding guidelines does not mandate it).